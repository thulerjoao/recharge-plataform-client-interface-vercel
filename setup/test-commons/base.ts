import type {Config} from '@jest/types'
import path from 'path'

//Sync object
const config: Config.InitialOptions = {
    testEnvironment: 'jsdom',
    verbose: true,
    setupFilesAfterEnv: [
        path.resolve(__dirname, 'src', 'setupTests.js')
    ],
    transform: {
        '\\.[jt]sx?$': ['babel-jest', {
            configFile: path.resolve(__dirname, '.babelrc')
        }],
    },
    moduleNameMapper: {
      '^axios$': require.resolve('axios'),
    },
    testMatch: [
      '**/__tests__/**/*.[jt]s?(x)',
      '**/?(*.)+(spec|test).[tj]s?(x)',
    ],
}
export default config

