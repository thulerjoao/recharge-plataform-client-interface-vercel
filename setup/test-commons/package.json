{
    "name": "@4miga/test-commons",
    "version": "1.0.0",
    "private": true,
    "main": "index.js",
    "scripts": {
      "lint": "echo 'Not necessary'",
      "test": "echo 'Not necessary'"
    },
    "files":[
      "base.ts"
    ],
    "devDependencies": {
        "jest":"27.0.0",
        "ts-node":"10.7.0",
        "babel-jest":"28.0.3",
        "@babel/core":"7.17.10",
        "@babel/preset-env":"7.17.10",
        "@babel/preset-react":"7.16.7",
        "@babel/preset-typescript":"7.16.7",
        "@testing-library/react": "14.2.1",
        "jest-styled-components" : "7.2.0"
    }
  }
