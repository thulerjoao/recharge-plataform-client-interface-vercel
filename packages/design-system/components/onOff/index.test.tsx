import { customRender } from "@4miga/test-commons/react-testing-library";
import OnOff from "./index";

const render = customRender();

describe("<OnOff /> snapshot tests", () => {
  it("onOff in on position", () => {
    const { container } = render(<OnOff onOff={true} />);
    expect(container).toMatchSnapshot();
  });

  it("onOff in off position", () => {
    const { container } = render(<OnOff onOff={false} />);
    expect(container).toMatchSnapshot();
  });
});
