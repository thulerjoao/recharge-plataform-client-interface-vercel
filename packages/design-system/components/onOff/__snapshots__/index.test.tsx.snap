// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OnOff /> snapshot tests onOff in off position 1`] = `
.c0 {
  width: 64px;
  min-width: 64px;
  height: 24px;
  border-radius: 12px;
  background-color: white;
  border: 1px solid #525F63;
  cursor: pointer;
  display: flex;
  align-items: center;
  transition: border 0.5s ease-in-out,box-shadow 0.5s ease-in-out;
}

.c0 .circle {
  height: 24px;
  width: 24px;
  border-radius: 50%;
  background-color: #525F63;
  transform: translateX(-1px);
  transition: transform 0.2s ease-in-out,background-color 0.5s ease-in-out;
}

<div>
  <div
    class="c0"
  >
    <div
      class="circle"
    />
  </div>
</div>
`;

exports[`<OnOff /> snapshot tests onOff in on position 1`] = `
.c0 {
  width: 64px;
  min-width: 64px;
  height: 24px;
  border-radius: 12px;
  background-color: white;
  box-shadow: 0px 0px 7px 0px #00C8FF;
  border: 1px solid #00C8FF;
  cursor: pointer;
  display: flex;
  align-items: center;
  transition: border 0.5s ease-in-out,box-shadow 0.5s ease-in-out;
}

.c0 .circle {
  height: 24px;
  width: 24px;
  border-radius: 50%;
  background-color: #00C8FF;
  transform: translateX(calc(40px));
  transition: transform 0.2s ease-in-out,background-color 0.5s ease-in-out;
}

<div>
  <div
    class="c0"
  >
    <div
      class="circle"
    />
  </div>
</div>
`;
